src/internal.rs

```diff
@@ -155,27 +158,30 @@ impl BitDepthDependentType for Grain {
 #[repr(C)]
 pub(crate) struct TaskThreadData_delayed_fg {
     pub exec: c_int,
-    pub cond: pthread_cond_t,
     pub in_0: *const Rav1dPicture,
     pub out: *mut Rav1dPicture,
     pub type_0: TaskType,
-    pub progress: [atomic_int; 2], /* [0]=started, [1]=completed */
     pub grain: BitDepthUnion<Grain>,
 }
 
+pub(crate) struct TaskThreadData_inner {
+    pub delayed_fg: TaskThreadData_delayed_fg,
+}
+
 #[repr(C)]
 pub(crate) struct TaskThreadData {
-    pub lock: pthread_mutex_t,
-    pub cond: pthread_cond_t,
+    pub lock: Mutex<TaskThreadData_inner>,
+    pub cond: Condvar,
     pub first: atomic_uint,
-    pub cur: c_uint,
+    pub cur: AtomicU32,
     /// This is used for delayed reset of the task cur pointer when
     /// such operation is needed but the thread doesn't enter a critical
     /// section (typically when executing the next sbrow task locklessly).
     /// See [`crate::src::thread_task::reset_task_cur`].
     pub reset_task_cur: atomic_uint,
     pub cond_signaled: atomic_int,
-    pub delayed_fg: TaskThreadData_delayed_fg,
+    pub delayed_fg_progress: [atomic_int; 2], /* [0]=started, [1]=completed */
```

Comment:
As I've refactored things, I've moved these kinds of comments to doc comments.  Not too important, but helpful and it seemed small and fine when I was already changing the field in the first place.